### ISSUE TITLE
How to use standard library h files (when dll built in Visual Studio)

### ISSUE THREAD

**Issue by mana2025** [analytech-solutions/ExamplesUsingC.jl#3]
Hi, 

I would like to thank you for the amazing CBinding package. It is very easy to work and very efficient. 

I am calling a dll which is from a c file a.c. This file has a header a.h. a.c calls several c files and h files. Some of those h files are c standard headers. For example string.h. I created new folder called "Myinclude" so that I can copy standard header files in it. Therefore, I included string.h as:
c"""
    #include "C:\Myinclude\string.h"
    #include "C:\Myinclude\a.h"
"""i
and works well. Now inside string.h there is a call for  #include <corecrt.h>
if I type the following
c"""
    #include "C:\Myinclude\corecrt.h"
"""i
I get the following error:

┌ Error: C:\Myinclude\string.h:12:10: error: 'corecrt.h' file not found with <angled> include; use "quotes" instead
└ @ CBinding C:\myuser\.julia\packages\CBinding\kBUap\src\context.jl:377

since #include <corecrt.h> is defined in string.h so I am reluctant to change <> to "". 
I do not refer neither to string.h nor to corecrt.h in the julia code. Is there a way to suppress these files so I do not have to include them from julia? 

Thanks a lot for your help.

**Comment by krrutkow** [comment]
You usually want to include standard headers as `#include <string.h>` to make sure the correct compiler-approved files are being used.  Sometimes headers do not include headers that define constructs that they reference, so you must include the header defining them yourself.  For that situation, you can include the header to satisfy the compiler yet skip the creation of bindings for it using the [`s` string macro option](https://github.com/analytech-solutions/CBinding.jl#options-for-c).
```julia
julia> c"""
         #include <stdio.h>  // provides FILE type, but skip emitting bindings for this block
       """s;

julia> c"""
         #include "header-that-forgot-to-include-stdio.h"  // header that just uses `FILE *` without including stdio.h
       """ij;
```

Is that what you're wanting to do?

**Comment by mana2025** [comment]
Thanks for your response. 

For me to understand better I am going to break this question to two: 

1- Where does the code think h files are located?
in my code the following does not give any error:

c"""
    #include <stdint.h>  
"""i;

but when I change the h file to string.h I get error:
c"""
    #include <string.h>  // provides FILE type, but skip emitting bindings for this block
"""i;
fatal error: 'string.h' file not found
│     #include <string.h>  // provides FILE type, but skip emitting bindings for this block     
│              ^~~~~~ here

Is that related to the dll I made (may be not made correctly) or it looks for it somewhere on the C drive? 

2- Let's say I included string.h successfully. But inside string.h there are other standard headers e.g., corecrt.h . If I use "I should not need to include corecrt.h unless there is a struct in my main header that being called but defined in corecrt.h. Is that right?

Thanks again for all your help.

**Comment by krrutkow** [comment]
1.  It seems that you are running on Windows, which would require you to have a C standard library installed.  It could come from something like MinGW, Visual Studio, etc.  But wherever it is installed will need to be specified for your CBinding compiler context to use it. `libclang` (used by CBinding) only provides some of the headers, and the others must be provided by your operating system.  You may need to use something like `--sysroot=<dir>` in your CBinding compiler context command to tell `libclang` where to find your C standard library.
2. That is correct!

**Comment by mana2025** [comment]
Thanks Krrutkow!

I added -I command with the path to headers and that resolved the error.
c`-std=c99 -Wall   -I$(path_to_headers)  -L$(mylib)  -lMydll.dll` 

Update on 1/21/2022: The above command got rid of string.h error but I was getting other errors regarding other headers. I took the following steps to resolve those errors:

1- copied all the headers needed to path_to_headers 

2- Since I am using visual studio 2017 I changed the following"
c`-fms-compatibility  -std=c99  -Wall -I$(path_to_headers)  -L$(mylib)  -lMydll.dll` 

I do not get any error now.
